### ISSUE TITLE
[feature-request] Allow ignoring errors by Error type

### ISSUE THREAD

**Issue by pedromss** [jarednorman/plugsnag#55]
I noticed that out of the box Plugsnag reports ~~404~~ `Phoenix.Router.NoRouteError` errors. I would like to request a feature that allows not reporting based on ~~status code~~ exception. I can't see a use case for wanting to report ~~404~~ `NoRouteError` as a bugsnag because anything can trigger a ~~404~~ `NoRouteError` so if someone were to spam our system with routes we don't have we'd essentially be bleeding out Bugsnag quota.

From the docs, there didn't seem to be a way around this, I only say the custom error builder, but I might have missed something.

Let me know if you're open to this and if you'd like me to take a shot at it.

**Comment by nirev** [comment]
Hi, @pedromss 

I've since moved to other providers so not actively using Bugsnag anymore.
I do think it makes sense and would welcome a PR!

**Comment by grzuy** [comment]
> I noticed that out of the box Plugsnag reports 404 errors

Does it?

Are changes introduced in https://github.com/bugsnag-elixir/plugsnag/pull/40 not working?

**Comment by pedromss** [comment]
You're correct, I made a mistake in the report. Its not a 404, its a `NoRouteError` error raised by Phoenix. I'll update the initial description to reflect that.

**Comment by grzuy** [comment]
Hey @pedromss ,

> You're correct, I made a mistake in the report. Its not a 404, its a `NoRouteError` error raised by Phoenix. I'll update the initial description to reflect that.

`NoRouteError` is a 404 per https://github.com/phoenixframework/phoenix/blob/e99f657f1cc9062fca0f2b8b79bc90659d8bd514/lib/phoenix/router.ex#L6, right?

So, I guess my question

> Are changes introduced in https://github.com/bugsnag-elixir/plugsnag/pull/40 not working?

still stands.
